javascript/windowing-system/windowing-system.js

```diff
@@ -31,33 +31,33 @@ export class ProgramWindow {
   }
 
   resize(newSize) {
-    let sizeW = newSize.width > 1 ? newSize.width : 1;
-    let sizeH = newSize.height > 1 ? newSize.height : 1;
-    const shiftSizeW = newSize.width + sizeW;
-    const shiftSizeH = newSize.height + sizeH;
+    let newWidth = newSize.width > 1 ? newSize.width : 1;
+    let newHeight = newSize.height > 1 ? newSize.height : 1;
+    const maxWidth = newSize.width + newWidth;
```

Comment:
This code is still broken by design.
```js
describe('move', () => {

  // ...

  test('fix me', () => {
    const programWindow = new ProgramWindow();
    const newPosition = new Position(programWindow.screenSize.width - 100, programWindow.screenSize.height - 100);

    programWindow.move(newPosition);
    const newSize = new Size(200, 200);
    programWindow.resize(newSize);

    expect(programWindow.size.width).toBe(100);
    expect(programWindow.size.height).toBe(100);
  });
});
```

Besides, this is NOT maximum width. Start thinking about what values mean.